java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java

```diff
@@ -139,18 +139,32 @@
                     Exceptions.printStackTrace(ex);
                     return CompletableFuture.completedFuture(true);
                 }
-                CompletableFuture<Project[]> projectsFuture = server.asyncOpenSelectedProjects(Collections.singletonList(file));
-                return projectsFuture.thenApply(projects -> {
-                    List<TestMethodController.TestMethod> testMethods = new ArrayList<>();
-                    for (Project prj : projects) {
-                        Set<URL> testRootURLs = new HashSet<>();
+                final Set<URL> testRootURLs = new HashSet<>();
+                return server.asyncOpenFileOwner(file).thenCompose(prj -> {
```

Comment:
Just in case: guard against `file == null`

Reply:
Fixed.